### PR TITLE
fix(response): return `errors` alongside `data`

### PR DESCRIPTION
**Pull request by sashman**
When the GraphQL throws an error and then responds with 200, it sends the errors in `errors` alongside `data`.  For example, an unauthenticated request might have this response:

```
{
  "data": null,
  "errors": [
    {
      "message": "Not authenticated",
      "locations": [
        {
          "line": 2,
          "column": 3
        }
      ],
      "path": [
        "..."
      ]
    }
  ]
}
```

BREAKING CHANGE

Now the value of `response.body` will be `%{ data: ..., errors: ... }` instead of `%{ ... }` (only returning `data`).

### CODE REVIEW

**Comment by uesteibar** [comment]
Hi @sashman!!
Good catch on this one. I tend to handle this via http status, but this usage should be supported as well. There's one test to be fixed though, could you take care of that?

Also, to comply with the [contributing guidelines](https://github.com/uesteibar/neuron#contributing) for this project, you'd need to change the commit message to something like:
```
fix(response): return `errors` alongside `data`
```
The commit descriptions could be something like:

```
BREAKING CHANGE

Now the value of `response.body` will be `%{ data: ..., errors: ... }` instead of `%{ ... }` (only returning `data`).
```

This allows for an easier-to-build [changelog](https://github.com/uesteibar/neuron/blob/master/CHANGELOG.md), that's why I prefer to stick to the convention.

Thanks again!

**Comment by uesteibar** [comment]
You could add the sample response you added in the commit description right before the `BREAKING CHANGE` part. I would rephrase the text to:

```
When the GraphQL throws an error and then responds with 200, it sends the errors in `errors` alongside `data`.  For example an unauthenticated request might have this response:

( here the sample response)

BREAKING CHANGE

...
```

**Comment by sashman** [comment]
My bad on missing the conventions! Sorry! :man_facepalming: Shall get onto it

**Comment by sashman** [comment]
I hope that's a bit better :)

**Comment by uesteibar** [comment]
Well done @sashman!! ❤️

**Comment by uesteibar** [comment]
@sashman this is now published on [v0.5.0](https://hex.pm/packages/neuron/0.5.0).

Great contribution, thanks! ❤️ 🎉
